crates/rustc_codegen_spirv/src/attr.rs

```diff
@@ -163,6 +166,10 @@ impl AggregatedSpirvAttributes {
             }
         }
 
+        if aggregated_attrs.capabilities.is_some() && aggregated_attrs.entry.is_none() {
+            cx.tcx.sess.err("#[spirv(capabilities)] can only be used with entry point functions.");
+        }
```

Comment:
For the record, this belongs at the end of `check_spirv_attributes`.

To avoid confusion around this in the future, maybe `AggregatedSpirvAttributes::parse` should be renamed to `AggregatedSpirvAttributes::gather` or something?